core/dbt/task/rpc/project_commands.py

```diff
@@ -97,3 +101,40 @@ def get_catalog_results(
             _compile_results=compile_results,
             logs=[],
         )
+
+
+class RemoteRunOperationTask(
+    RPCTask[RPCRunOperationParameters],
+    HasCLI[RPCRunOperationParameters, RemoteRunOperationResult],
+    RunOperationTask,
+):
+    METHOD_NAME = 'run-operation'
+
+    def set_args(self, params: RPCRunOperationParameters) -> None:
+        self.args.macro = params.macro
+        self.args.args = params.args
+
+    def _get_kwargs(self):
+        if isinstance(self.args.args, dict):
+            return self.args.args
+        else:
+            return RunOperationTask._get_kwargs(self)
+
+    def _runtime_initialize(self):
+        return RunOperationTask._runtime_initialize(self)
+
+    def handle_request(self) -> RemoteRunOperationResult:
+        success, _ = RunOperationTask.run(self)
+        result = RemoteRunOperationResult(logs=[], success=success)
+        return result
+
+    def interpret_results(self, results):
+        return results.success
+
+
+class RemoteSnapshotTask(RPCCommandTask[RPCCompileParameters], SnapshotTask):
+    METHOD_NAME = 'snapshot'
+
+    def set_args(self, params: RPCCompileParameters) -> None:
+        self.args.models = self._listify(params.models)
```

Comment:
I think the input to `snapshot` on the CLI is `-s --select`, not `--models` - can we update this accordingly?